### PR TITLE
19 profiles beyond nfw

### PR DESCRIPTION
**Pull request by LanceM28**
Addition of the einasto profile with density, enclosed mass, circular velocity, and gravitational potential functions. Note that the numerical potential is normalized to a finite value and may fail a modified test at larger radii.

test_nfw.py now includes gala comparison for independent validation of nfw gravitational potential.

### CODE REVIEW

File: src/halox/halo/einasto.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        Returns
+        -------
+        Array [km s-1]
+            Circular velocity at radius `r`
+        """
+        r = jnp.asarray(r)
+        return jnp.sqrt(G * self.enclosed_mass(r) / r)
+
+    def potential(self, r: ArrayLike) -> Array:
+        # NO TEST YET, autodiff compatability with incomplete gamma function
```
Comment (fkeruzore) [inline src/halox/halo/einasto.py]: then it might be interesting to add this function to the jax friendliness tests in the future!

File: tests/test_einasto.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+    if return_vals:
+        return res_h, res_c
+
+    assert jnp.allclose(jnp.array(res_c), res_h, rtol=rtol), (
+        f"Different v_circ({rs}): {res_c} != {res_h}"
+    )
+
+
+# @pytest.mark.parametrize("halo_name", test_halos.keys())
```
Comment (fkeruzore) [inline tests/test_einasto.py]: do we want to keep these commented out functions?

File: tests/test_nfw.py [hunk 3]
```diff
@@ -99,33 +102,35 @@ def test_enclosed_mass(
 @pytest.mark.parametrize("halo_name", test_halos.keys())
 @pytest.mark.parametrize("delta", test_deltas)
 @pytest.mark.parametrize("cosmo_name", test_cosmos.keys())
-def test_potential(halo_name, delta, cosmo_name, return_vals: bool = False):
+def test_potential(
+    halo_name, delta, cosmo_name, return_vals: bool = False
+):  # can I remove need for astropy.units???
+    kmperkpc = 30856775999999956
```
Comment (fkeruzore) [inline tests/test_nfw.py]: since you're already importing Astropy units, I'd use that instead of hard-coding conversions

File: pyproject.toml
Comment (fkeruzore) [inline pyproject.toml]: gala shouldn't be a main dependency, I'll put it in groups

**Review by fkeruzore — CHANGES_REQUESTED** [review]
looking really good; mostly just a few questions about comments and whether or not they should stay in, and I made some minor tweaks to the colossus comparison notebook

File: src/halox/halo/einasto.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    z: float
+        Redshift
+    cosmo: jc.Cosmology
+        Underlying cosmology
+    delta: float
+        Density contrast in units of critical density at redshift z,
+        defaults to 178.
+    """
+
+    # Lance Questions
```
[the remark this answers is not in the document]
Comment (LanceM28) [reply src/halox/halo/einasto.py]: I think I am going to get rid of most of them, they were earlier notes that I just forgot to delete so i'll get rid of the ones I no longer need and are not necessary for future things.

**Review by fkeruzore — APPROVED** [review]
